tests/foreman/longrun/test_oscap.py

```diff
@@ -598,10 +718,34 @@ def test_positive_oscap_run_via_local_files(
 
     :CaseImportance: Critical
     """
+    contenthost = rex_contenthost
+    os_version = contenthost.os_version.major
+    distro = f'rhel{os_version}'
     SELECTED_ROLE = 'theforeman.foreman_scap_client'
     file_name = 'security-data-oval-com.redhat.rhsa-RHEL8.xml.bz2'
     download_url = 'https://www.redhat.com/security/data/oval/v2/RHEL8/rhel-8.oval.xml.bz2'
     profile = OSCAP_PROFILE['ospp8']
+
+    # list existing content on Sat
+    existing_content_names = get_existing_content(module_target_sat)
+    # the content name depends on upload style (default vs file upload)
+    if (OSCAP_DEFAULT_CONTENT[f'{distro}_content'] not in existing_content_names) and (
+        f'{distro} content' not in existing_content_names
+    ):
+        upload_scap_content_for_host(module_target_sat, contenthost, module_org)
+
+    # re-list after content upload
+    existing_content_names = get_existing_content(module_target_sat)
+    # find out content name for client version
+    assert (OSCAP_DEFAULT_CONTENT[f'{distro}_content'] in existing_content_names) or (
+        f'{distro} content' in existing_content_names
+    ), f'scap content not found for {distro}, existing content: {existing_content_names}'
+    content = (
+        OSCAP_DEFAULT_CONTENT[f'{distro}_content']
+        if (OSCAP_DEFAULT_CONTENT[f'{distro}_content'] in existing_content_names)
+        else f'{distro} content'
+    )
```

Comment:
This looks duplicated from `test_positive_oscap_run_via_ansible_bz_1814988`. Would it make sense to factor it out to a helper function?